src/gui/editorwidgets/qgstexteditwrapper.cpp

```diff
@@ -74,6 +75,10 @@ QVariant QgsTextEditWrapper::value() const
     // input rather then discarding it entirely
     return QVariant( v.left( field().length() ) );
   }
+  else if ( field().type() == QVariant::Map )
+  {
+    return v;
```

Comment:
Here the parsing should happen

Reply:
What if the JSON parsing fails? I think this currently returns NULL, but that sounds like it's too easy for a user to kill his data.

Reply:
If json parsing fails it defaults back to text, so the field will contain a JSON text string as a single JSON object.